src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeUtils.cs

```diff
@@ -180,6 +180,10 @@ public static bool IsValidInstanceType(MemberInfo member, Type instanceType)
             {
                 return true;
             }
+            if (targetType == null)
```

Comment:
`it ((object)targetType == null)`   -   just to make it cheaper by not using the "==" operator on Type